common/writeThroughFile_freebsd.go

```diff
@@ -0,0 +1,61 @@
+// +build !386
```

Comment:
I don't recall exactly why I added `!i386` in particular, but yes, I believe the code is not functional on 32-bit platforms.

Go's `Syscall()` function is defined as taking `uintptr`-sized parameters.  If Go's `uintptr` matches the C type, that's 32-bit on 32-bit FreeBSD architectures.  But `fileSize` (and FreeBSD's `posix_fallocate(2)` / `SYS_POSIX_FALLOCATE` system call's 3rd parameter) is explicitly a 64-bit size.  So I didn't want to suggest the implementation was correct on i386 — certainly it's wrong today, in that `uintptr(fileSize)` would truncate the allocation requested for files longer than 4GB.

It would be nice, and would avoid this problem, if Go's FreeBSD `Syscall` package had `Fallocate()` (or `PosixFallocate`, but they are the same thing) already defined, like Linux's.  But it does not, and AIUI has been frozen.  Unfortunately, the new extended `x/sys/unix` package doesn't provide FreeBSD's `posix_fallocate` either: https://godoc.org/golang.org/x/sys/unix

i386 isn't the only 32-bit FreeBSD platform, so I'm not sure why I only marked that one.  Maybe azcopy only supports amd64 and i386?  If so, that may be why.